### ISSUE TITLE
mypy error: "FastAPI" has no attribute "container"

### ISSUE THREAD

**Issue by SpyrosRoum** [ets-labs/python-dependency-injector#626]
dependecy-injector version: "^4.40.0"
FastAPI version: "^0.79.1"

Hello, following the example for FastAPI, mypy complains with the error on the title when I do `app.container = container`.

The only way I found to work around it (other than `type: ignore`) was to create a custom class that inherits from FastAPI:

```
class MyFastAPI(FastAPI):
    container: Container


def create_app() -> MyFastAPI:
    app = MyFastAPI()

    container = Container()
    app.container = container

    @app.on_event('startup')
    async def startup_event() -> None:
        app.container.wire(modules=[__name__])
        await initialize(app.container)

    return app
```

Is there any other way to do this?
I'm a little surprised that I couldn't find any other issue about this so I'm wondering if I missed something.

**Comment by ffss92** [comment]
I mean, is that really a work around? You needed to add a new property to the FastAPI class so you extended it. Seems like a normal solution.

**Comment by SpyrosRoum** [comment]
You are right that this is a fairly good solution, I'm just mostly surprised that this was not mentioned anywhere and I wasn't able to find any issues or anything else about it.

For a lib that claims to be mypy friendly I'd expect it to mention something instead of just giving an example that doesn't actually work with mypy

**Comment by ffss92** [comment]
FastAPI is a third party package though, I think 'mypy fiendly' means that the lib provides it's own types.

**Comment by ZipFile** [comment]
It doesn't, mypy is right here. There is no special attribute-checking logic in python-dependency-injector neither FastAPI has `container` attribute. More idiomatic way would be something like #683.
